server/resources/resource_management_test.go

```diff
@@ -1438,12 +1438,78 @@ var _ = Describe("Resource manager", func() {
 							"test_string": "Steloj ne estas en ordo.",
 							"test_bool":   true,
 						}
-						events["pre_update"] = `if (context.resource.test_bool === undefined) { context.resource.test_bool = false; }`
+						events["pre_create"] = `if (context.run_precreate) { context.resource.test_string = "test123"; }`
 						context["go_validation"] = true // simulate Goext
 						context["request_data"] = requestMap
 					})
+					It("should create non-empty resource", func() {
+						context["request_data"] = deepcopy.Copy(requestMap).(map[string]interface{})
+
+						err := resources.CreateResource(context, testDB, fakeIdentity, currentSchema, requestMap)
+						result := context["response"].(map[string]interface{})
+						Expect(err).NotTo(HaveOccurred())
+						theResource, ok := result[schemaID]
+						Expect(ok).To(BeTrue())
+						Expect(theResource).To(HaveKeyWithValue("test_bool", true))
+						Expect(theResource).To(HaveKeyWithValue("test_string", "Steloj ne estas en ordo."))
+					})
+
+					It("should create empty resource", func() {
+						delete(requestMap, "test_bool")
+						delete(requestMap, "test_string")
+						context["request_data"] = deepcopy.Copy(requestMap).(map[string]interface{})
+
+						err := resources.CreateResource(context, testDB, fakeIdentity, currentSchema, requestMap)
+						result := context["response"].(map[string]interface{})
+						Expect(err).NotTo(HaveOccurred())
+						theResource, ok := result[schemaID]
+						Expect(ok).To(BeTrue())
+						Expect(theResource).To(HaveKeyWithValue("test_bool", false))
+						Expect(theResource).To(HaveKeyWithValue("test_string", ""))
+					})
```

Comment:
You should include an unit test of failing resource creation if required value has not been filled in Goext.